low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,369 @@
+        self.boot_drvs_dta = "{0}/Server_OS_boot_drives_SMART_data_{1}".format(
+            sspl_const.SSPL_SB_TMP, str(int(time.time())))
+        self.enclosure_log = f"{sspl_const.SSPL_SB_TMP}/enclosure_logs.zip"
+        self.enclosure_wwn = "NA"
+        self.sspl_log_dir = Conf.get(SSPL_CONF, "%s>sspl_log_file_path" %
+                                     (self.SYS_INFORMATION)).replace("/sspl.log", "")
+        self.iem_log_dir = Conf.get(SSPL_CONF, "%s>log_file_path" %
+                                    (self.IEM_SENSOR)).replace("/iem_messages", "")
+
+    def generate(self, args):
+        """Generate support bundle."""
+        logger.info("Starting SSPL bundle generation")
+        self.prepare_bundle(args.bundle_id, args.target_path, args.noencl)
+        self.__clear_tmp_files()
+        status = SupportBundle().get_status(args.bundle_id)
+        print(status)
+
+    @staticmethod
+    def get_status(bundle_id: str):
+        """Get status of support bundle."""
+        status = None
+        # req_id = "sb_%s" % bundle_id
+        # sspl_sb_id = req_register.get(["%s>sspl_sb_id" % req_id])
+        request_status = BundleRequestHandler.get_processing_status(bundle_id)
+        # Extract sspl sb request status from the message.
+        # Example : "Failed - error(22): Invalid request"
+        status = request_status.split(" ")[0]
+        if status != "In-progress":
+            return status
+        return None
+
+    def prepare_bundle(self, bundle_id: str, target_path: str, include_encl):
+        """Collect sspl support bundle data."""
+        bundle_name = "sspl_" + bundle_id
+        request_id = datetime.strftime(datetime.now(), '%Y%m%d%H%M%S%f')
+        try:
+            self.get_sspl_bundle_data(bundle_id, request_id, target_path, include_encl)
+            bundle_location = self._generate_tar(bundle_name)
+            SupportBundle._update_bundle_request_info(
+                bundle_id, request_id, bundle_location)
+            BundleRequestHandler._update_request_status(
+                bundle_id, BundleRequestHandler.SUCCESS)
+        except Exception as err:
+            status = BundleRequestHandler.FAILED + f" - {err}"
+            BundleRequestHandler._update_request_status(
+                bundle_id, status)
+
+    def get_sspl_bundle_data(self, bundle_id, request_id, file_path, include_encl):
+        """Collect SSPL data for support bundle."""
+        BundleRequestHandler._add_bundle_request(bundle_id, request_id, file_path)
+
+        if file_path:
+            if os.path.exists(file_path):
+                file_path = file_path+"/sspl/"
+                self.tmp_bundle_path = os.path.normpath(file_path)
+            else:
+                msg = "Given path %s doesn't exist" % (file_path)
+                self.__clear_tmp_files()
+                raise SupportBundleError(1, msg)
+        os.makedirs(self.tmp_bundle_path, exist_ok=True)
+        try:
+            self.get_ipmi_sel_data()
```

Comment:
Can we put these steps in to different try catch blocks, so exception can become specific too and generation can continue for rest of the data ?

Reply:
Done